crawl-ref/source/dat/des/branches/swamp.des

```diff
@@ -2361,3 +2361,67 @@ ttccccC'''cC'-'Cc'''1'1cctt
      ttttttttttttttttt t
        tt   ttt ttt t
 ENDMAP
+
+###############################################################################
+# Growth, Death and Undeath, by byrel
+# 1/11 chance of an undead lernaean hydra
+NAME:    byrel_swamp_growth_death_and_undeath
+TAGS:    no_monster_gen
+PLACE:   Swamp:$
+ORIENT:  float
+TAGS:    no_rotate
+MONS:    swamp dragon / swamp drake / hydra
+MONS:	 freezing wraith / flayed ghost / phantasmal warrior / ghoul
+MONS:    hydra skeleton / swamp dragon skeleton / death drake skeleton
+MONS:    hydra zombie / swamp dragon zombie / death drake zombie
+MONS:    spectral hydra / hydra simulacrum
+MONS:    patrolling thorn hunter / patrolling shambling mangrove
+MONS:    patrolling spriggan druid god:fedhas
+KITEM:   O = decaying rune of Zot
+KFEAT:   _ = altar_fedhas
+KFEAT:   D = altar_yredelemnul
+NSUBST:  O = 1:O / *:%
+SUBST:   x = xxxxxxttT
+SUBST:   1 = 1.W
+NSUBST:  3 = 3=5 / 3..
+SUBST:   3 = 2344
+SUBST:   . = ...W
+NSUBST:  P = 3=' / ppW
+KMONS:   p = plant
+KMONS:   ' = oklob plant
+KMONS:   q = vampire knight
+: if crawl.one_chance_in(11) then
```

Comment:
A few questions here. This vault will occur in like 7% of Swamp:$ instances, given how many other Swamp:$ vaults there are. Placing the undead Learnaean hydra in just 9% of those will make it a pretty rare find. I imagine that's intentional, I'm just a little surprised you wanted it to be so rare. Do you think it's too difficult to place 100% of the time? Do you want it to be so rare as to be very exciting when a player runs into it?

Second, I wouldn't do this with a one_chance_in. Why not just do it with a weighted MONS, where the sum of weights for undead lernaean hydrae is much less than the total weight?

    MONS: the lernaean hydra skeleton / spectral the lernaean hydra / ... / spectral hydra w:100 / hydra simulacrum w:100

Reply:
For both of these, I was copying swamp_basic_johnstein; now that I look more carefully, I see that one is an outlier. I'm pretty sure undead lerny is significantly easier than non-undead-lerny; low MR, and the only high damage one is also slow.

I'll change this.